client/rest-high-level/src/test/java/org/elasticsearch/client/ml/FlushJobRequestTests.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.ml;
+
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.test.AbstractXContentTestCase;
+
+import java.io.IOException;
+
+public class FlushJobRequestTests extends AbstractXContentTestCase<FlushJobRequest> {
+
+    @Override
+    protected FlushJobRequest createTestInstance() {
+        FlushJobRequest request = new FlushJobRequest(randomAlphaOfLengthBetween(1, 20));
+
+        if (randomBoolean()) {
+            request.setCalcInterim(randomBoolean());
+        }
+        if (randomBoolean()) {
+            request.setAdvanceTime(String.valueOf(randomLong()));
+        }
+        if (randomBoolean()) {
+            request.setStart(String.valueOf(randomLong()));
+        }
+        if (randomBoolean()) {
+            request.setEnd(String.valueOf(randomLong()));
+        }
+        if (randomBoolean()) {
+            request.setSkipTime(String.valueOf(randomLong()));
+        }
+        return request;
+    }
+
+    @Override
+    protected FlushJobRequest doParseInstance(XContentParser parser) throws IOException {
+        return FlushJobRequest.PARSER.apply(parser, null);
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
+        return true;
```

Comment:
The request should not be supporting unknown fields.

Reply:
I've noticed many of the request and response objects support unknown fields I thought this was by design. For the requests the only time they are parsed is in the unit tests so that seems redundant. Responses are parsed by the client and if the client is not the same version as the server is may see unknown fields in which case we should error as nothing is learned by silently ignoring the field.

Reply:
We need to leniently parse responses to enable compatibility of the HLRC client with multiple versions. Imagine you use rest directly and we add some new fields in a response that your script is simply ignoring. It is the same. On the request objects, on the other hand, we don't need to be lenient. In fact, for those it doesn't matter as they only have a parser for testing but there is no point doing the extra work of randomizing fields in the tests.

Reply:
> For the requests the only time they are parsed is in the unit tests so that seems redundant

True

> Responses are parsed by the client and if the client is not the same version as the server is may see unknown fields in which case we should error

A 7.0 client needs to be compatible with 7.6, and between 7.0 and 7.6 we may have added extra fields to the responses.  So I think the HLRC _should_ ignore unknown fields in responses.  Obviously by using a 7.0 client with 7.6 you lose out on any new functionality, but we should not have broken backwards compatibility so missing out on the new fields shouldn't break anything that worked in 7.0.  #11184 is related to this, although I think any changes will be too late to affect what we do in these classes in the short term.